solidity/contracts/Gravity.sol

```diff
@@ -178,11 +179,13 @@ contract Gravity is ReentrancyGuard {
 		uint8 _v,
 		bytes32 _r,
 		bytes32 _s
-	) private pure returns (bool) {
+	) private pure returns (bool ) {
 		bytes32 messageDigest = keccak256(
 			abi.encodePacked("\x19Ethereum Signed Message:\n32", _theHash)
 		);
-		return _signer == ecrecover(messageDigest, _v, _r, _s);
+		(address recAddr, ) = ECDSA.tryRecover(messageDigest, _v, _r, _s);
+		require(recAddr != address(0), "zero ecrec address");
```

Comment:
I think that this check is made in the `tryRecover` funtion. could you double check the implementation

Reply:
Yes, the check is done already inside tryEcrecover, so this will be removed.